Sources/Sentry/SentryCrashReportConverter.m

```diff
@@ -46,7 +46,21 @@ - (instancetype)initWithReport:(NSDictionary *)report
 
         self.diagnosis = crashContext[@"diagnosis"];
         self.exceptionContext = crashContext[@"error"];
-        self.threads = crashContext[@"threads"];
+        [self initThreads:crashContext[@"threads"]];
+    }
+    return self;
+}
+
+- (void)initThreads:(NSArray<NSDictionary *> *)threads
+{
+    if (nil != threads && [threads isKindOfClass:[NSArray class]]) {
+        // SentryCrash sometimes produces recrash_reports where an element of threads is a
+        // NSString instead of a NSDictionary. When this happens we can't read the details of
+        // the thread, but we have to discard it. Otherwise we would crash.
+        NSPredicate *onlyNSDictionary = [NSPredicate predicateWithBlock:^BOOL(id object,
```

Comment:
L: is this formatted by clang-format? it's odd to have the `{ ... }` inlined like that IMHO but this is a nit

Reply:
Yep that was clang. Thanks, for pointing it out. I changed it.